### PR TITLE
[Address]

### PR DESCRIPTION
**Pull request by PapsOu**

### CODE REVIEW

File: doc/fr/Sil/Component/Address/domain/index.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+| city            | La ville                                              | x      |
++-----------------+-------------------------------------------------------+--------+
+| other           | Un complément d'adresse, nécessaire pour certains cas |        |
++-----------------+-------------------------------------------------------+--------+
+
+
+Ville
+-----
+
+Chaque **Ville** peut appartenir à un pays. On permet la saisie de ville sans forcément avoir le pays associé. À charge de chaque composant de rendre plus strict cette gestion.
```
Comment (RomainSanchez) [inline doc/fr/Sil/Component/Address/domain/index.rst]: stricte

File: doc/fr/Sil/Component/Address/domain/index.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
++-----------------+------------------------------------------------+--------+
+| postCode        | Un code postal                                 | x      |
++-----------------+------------------------------------------------+--------+
+| code            | Un code (ex: code INSEE) représentant la ville |        |
++-----------------+------------------------------------------------+--------+
+
+Code postal
+-----------
+
+Un **Code Postal** est un object comprenant un code sous forme de chaîne (pour permettre les code commençants par 0xxxx) et de longueur non fixée pour permettre la gestion de code postaux étrangers.
```
Comment (RomainSanchez) [inline doc/fr/Sil/Component/Address/domain/index.rst]: "codes" and "codes postaux"

File: src/Sil/Component/Address/Repository/CityRepositoryInterface.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    /**
+     * Retreive the city for specific code.
+     *
+     * @param string $code
+     *
+     * @return CityInterface
+     *
+     * @throws InvalidArgumentException
+     */
+    public function getByCode(string $code): CityInterface;
```
Comment (RomainSanchez) [inline src/Sil/Component/Address/Repository/CityRepositoryInterface.php]: findByCode ?

File: doc/fr/Sil/Component/Address/domain/index.rst [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
++=================+========================================+========+
+| name            | Le nom du pays                         | x      |
++-----------------+----------------------------------------+--------+
+| code            | Le code pays normalisé (voir ISO 3166) | x      |
++-----------------+----------------------------------------+--------+
+
+Provinces
+---------
+
+Les provinces ne sont pas présentes dans ce composant étant donnée que leur définition dépend entièrement des cas d'usages. Ce composant n'offre donc pas par défaut de gestion de région / département / autre groupement administratif.
```
Comment (RomainSanchez) [inline doc/fr/Sil/Component/Address/domain/index.rst]: "donné"

File: src/Sil/Component/Address/Repository/CityRepositoryInterface.php (hunk 3 above)
Comment (PapsOu) [reply src/Sil/Component/Address/Repository/CityRepositoryInterface.php]: Diffrence between `get` and `find` would be : find tries to find an element, so the method would return a model or a null result, a get must return a model. If not, it's an Exception (InvalidargumentException usualy)

File: src/Sil/Component/Address/Model/CountryInterface.php [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+     * @return string
+     */
+    public function getCode(): string;
+
+    /**
+     * Sets the city code.
+     *
+     * @param string $code
+     */
+    public function setCode(string $code): void;
```
Comment (RomainSanchez) [inline src/Sil/Component/Address/Model/CountryInterface.php]: Why setCode in the interface when code is already a mandatory parameter of the constructor

File: src/Sil/Component/Address/Model/Address.php [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+     */
+    public function getOther(): ?string
+    {
+        return $this->other;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setOther(string $other): void
```
Comment (GlennCavarle) [inline src/Sil/Component/Address/Model/Address.php]: should be `?string $other` if getOther() can return null.

File: src/Sil/Component/Address/Model/Country.php [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+/*
+ * Copyright (C) 2015-2018 Libre Informatique
+ *
+ * This file is licenced under the GNU LGPL v3.
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Sil\Component\Address\Model;
+
```
Comment (GlennCavarle) [inline src/Sil/Component/Address/Model/Country.php]: `use InvalidArgumentException; `is missing

File: doc/fr/Sil/Component/Address/domain/index.rst [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+| city            | La ville                                              | x      |
++-----------------+-------------------------------------------------------+--------+
+| other           | Un complément d'adresse, nécessaire pour certains cas |        |
++-----------------+-------------------------------------------------------+--------+
+
+
+Ville
+-----
+
+Chaque **Ville** peut appartenir à un pays. On permet la saisie de ville sans forcément avoir le pays associé. À charge de chaque composant de rendre cette gestion plus ou moins stricte .
```
Comment (GlennCavarle) [inline doc/fr/Sil/Component/Address/domain/index.rst]: But it is not actually true, the City is mandatory in the source code and in the table below.

**Comment by GlennCavarle** [comment]
can be merged after minor revisions.

File: src/Sil/Component/Address/Model/Address.php (hunk 6 above)
Comment (PapsOu) [reply src/Sil/Component/Address/Model/Address.php]: Hum, setting a null value isn't very logical... The interface tell that the value should not be initialized when creating object.
Maybe adding a `clearOther(): void` method to clear the value would be more semantic ?

File: doc/fr/Sil/Component/Address/domain/index.rst (hunk 8 above)
Comment (PapsOu) [reply doc/fr/Sil/Component/Address/domain/index.rst]: Yes I forgive to change this part of the doc

**Pull request merged**